### PR TITLE
copy-paste from gitlab 

### PR DESCRIPTION
**Pull request by glyg**
Code was previously here

https://gitlab.in2p3.fr/fbi-data/small-and-various-pieces-of-coding/omero-webapp-cookiecutter

### CODE REVIEW

**Comment by will-moore** [comment]
Just tested this with

```
$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy
```
and it worked fine.

```
$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy
  [1/7] full_name (Guillaume Gay): Joe Blogs
  [2/7] email (guillaum@damcb.com): j.blogs@example.com
  [3/7] webapp_name (simple_webapp): test_app
  [4/7] webapp_title (Simple Webapp): Test App
  [5/7] description (OMERO example minimal web app): just testing
  [6/7] repo_name (omero-simple-webapp): omero-test-webapp
  [7/7] version (0.1.0): 
```

I would like to acknowledge your contribution @glyg, but maybe your name and e-mail shouldn't be the default values. In fact I'm not sure there should be default values for these since we always want users to enter something. Does it work without defaults?

I think we can tweak the README of this app (and the generated app) in follow-up PRs, but it would be good to fix the cookiecutter instruction in this PR:

```
cookiecutter https://gitlab.in2p3.fr/fbi-data/small-and-various-pieces-of-coding/omero-webapp-cookiecutter
```
should be
```
cookiecutter https://github.com/ome/cookiecutter-omero-webapp.git
```

Thanks!

**Comment by glyg** [comment]
I don't think there's reason for the default email, will look into it. No worries about recognition :)

**Comment by glyg** [comment]
I'll fix the url later today

**Comment by glyg** [comment]
I think the json is written as an update to defaults, it shouldn't have been versioned

**Review by will-moore — APPROVED** [review]
Great, thanks

**Comment by pwalczysko** [comment]
```
(cookicutter) pwalczysko@Petrs-MBP cookiecutter-omero-webapp % cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy
You've downloaded /Users/pwalczysko/.cookiecutters/cookiecutter-omero-webapp 
before. Is it okay to delete and re-download it? [y/n] (y): y
A valid repository for "https://github.com/glyg/cookiecutter-omero-webapp.git" could not be found in the following locations:
/Users/pwalczysko/.cookiecutters/cookiecutter-omero-webapp
```

Not sure ^^^ what is missing. Tried to follow the simple README:
- cloned this repo, ``cd`` into it
- checked out this branch
- Created a new conda env cookcutter ``conda create -n cookicutter python=3.10``
- activated the env
- ``pip install cookiecutter``
- ``cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy`` -> ends with error above

**Comment by will-moore** [comment]
That's weird - don't know what's going on there. I get the same thing. I even tried deleting the cached repo, but it didn't help. Also tried to use the earlier commit that worked for me above, but no joy.

```
$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout f1d4703
You've downloaded /Users/wmoore/.cookiecutters/cookiecutter-omero-webapp before. Is it okay to delete and re-download it? [y/n] (y): y
A valid repository for "https://github.com/glyg/cookiecutter-omero-webapp.git" could not be found in the following locations:
/Users/wmoore/.cookiecutters/cookiecutter-omero-webapp

$ rm -rf /Users/wmoore/.cookiecutters/cookiecutter-omero-webapp

$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy
A valid repository for "https://github.com/glyg/cookiecutter-omero-webapp.git" could not be found in the following locations:
/Users/wmoore/.cookiecutters/cookiecutter-omero-webapp
```

**Comment by will-moore** [comment]
@glyg - I think you need to add back the `cookiecutter.json` as it look like that's essential.
I added it to my local `/Users/wmoore/.cookiecutters/cookiecutter-omero-webapp` cache, and I was then able to use that...

```
$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git
You've downloaded /Users/wmoore/.cookiecutters/cookiecutter-omero-webapp before. Is it okay to delete and re-download it? [y/n] (y): n
Do you want to re-use the existing version? [y/n] (y): y
  [1/7] full_name (Guillaume Gay): Will Moore
...
```
We can update defaults in a follow-up PR. Thanks.

File: {{cookiecutter.repo_name}}/{{cookiecutter.webapp_name}}/views.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from django.shortcuts import render
+from django.core.urlresolvers import reverse
```
Comment (will-moore) [inline {{cookiecutter.repo_name}}/{{cookiecutter.webapp_name}}/views.py]: reverse isn't used and this import actually fails on Django 4.2 so this line can be removed

File: {{cookiecutter.repo_name}}/{{cookiecutter.webapp_name}}/urls.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from django.conf.urls import url
```
Comment (will-moore) [inline {{cookiecutter.repo_name}}/{{cookiecutter.webapp_name}}/urls.py]: This import fails now on Django 4.2. It needs to be `from django.urls import re_path` and then replace `url()` with `re_path()` below.

**Review by will-moore — CHANGES_REQUESTED** [review]
Thanks @glyg - that's working again.
I found a couple of issues I missed before when actually installing the app

**Comment by will-moore** [comment]
Testing...

```
$ cookiecutter https://github.com/glyg/cookiecutter-omero-webapp.git --checkout legacy
You've downloaded /Users/wmoore/.cookiecutters/cookiecutter-omero-webapp before. Is it okay to delete and re-download it? [y/n] (y): y
  [1/7] full_name (Example): Wills_App       
  [2/7] email (login@example.com): 
  [3/7] webapp_name (simple_webapp): wills_app
  [4/7] webapp_title (Simple Webapp): Wills App
  [5/7] description (OMERO example minimal web app): test app
  [6/7] repo_name (omero-simple-webapp): omero-wills-app
  [7/7] version (0.1.0): 

$ omero config append omero.web.apps '"wills_app"'
$ ls
omero-wills-app
$ cd omero-wills-app/
$ pip install -e .
Obtaining file:///Users/wmoore/Desktop/COOKIECUTTER/omero-wills-app
  Preparing metadata (setup.py) ... done
Installing collected packages: omero-wills-app
  Running setup.py develop for omero-wills-app
Successfully installed omero-wills-app-0.1.0
```

Restarted omero-web server and went to http://localhost:4080/wills_app/

Screenshot 2024-02-07 at 08 55 16

**Review by will-moore — APPROVED** [review]
Looks good, thanks 👍

File: README.md [hunk 3]
```diff
@@ -1,2 +1,18 @@
-# cookiecutter-omero-webapp
-Cookiecutter to create OMERO.web apps
+# Cookiecutter to create simple OMERO webapps
+
+[Cookiecutter](https://cookiecutter.readthedocs.io) allows to create a project from a predefined template, taking care of all the boilerplate code and filling in the relevant information from user's input.
+
+
+
+
+This repository offers a  for OMERO web app by creating a project skeleton. It is based on the Django example from here: https://github.com/ome/omero-web-apps-examples
```
Comment (joshmoore) [inline README.md]:
```suggestion
This repository offers a template for an OMERO web app by creating a project skeleton.
It is based on the Django example from https://github.com/ome/omero-web-apps-examples
```

File: {{cookiecutter.repo_name}}/setup.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from setuptools import setup, find_packages
+
+setup(
+    name="{{cookiecutter.repo_name}}",
+    version="{{cookiecutter.version}}",
+    description="{{cookiecutter.description}}",
+    packages=find_packages(),
+    keywords=["omero", "example"],
```
Comment (joshmoore) [inline {{cookiecutter.repo_name}}/setup.py]:
```suggestion
    keywords=["omero"],
```

File: {{cookiecutter.repo_name}}/README.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
+OMERO.{{cookiecutter.webapp_name}}
+=======================
+
+This repository contains an
```
Comment (joshmoore) [inline {{cookiecutter.repo_name}}/README.md]:
```suggestion
This repository contains a
```

File: {{cookiecutter.repo_name}}/README.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+[cookiecutter](https://cookiecutter.readthedocs.io/) generated example
+of a minimal omero webapp, derived from openmicroscopy [webapp examples](https://github.com/ome/omero-web-apps-examples).
```
Comment (joshmoore) [inline {{cookiecutter.repo_name}}/README.md]:
```suggestion
of a minimal OMERO webapp, derived from openmicroscopy [webapp examples](https://github.com/ome/omero-web-apps-examples).
```

File: {{cookiecutter.repo_name}}/README.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+This template repository contains a minimal example of OMERO.web apps minimal example (no JavaScript dependencies or build steps).
```
Comment (joshmoore) [inline {{cookiecutter.repo_name}}/README.md]:
```suggestion
This repository contains a minimal OMERO.web app (no JavaScript dependencies or build steps).
```

**Review by joshmoore — COMMENTED** [review]
Few minor edits in the README, but we can also get this in and then iterate on that.

Thanks, @glyg! I look forward to trying it out.

**Comment by will-moore** [comment]
Thanks all!

**Pull request merged**